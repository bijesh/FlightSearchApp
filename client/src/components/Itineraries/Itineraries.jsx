import React from 'react';
import Itinerary from '../Itinerary';
import {uid} from 'react-uid';
import STYLES from './Itineraries.scss';

const ItinerariesStyle = className => STYLES[className] || 'UNKNOWN';

const Itineraries = React.memo(({ itineraryDetails }) => (

    itineraryDetails.map(item => (
        <Itinerary key={uid(item)} itinerary={item}/>
    ))

    ));

export default Itineraries;