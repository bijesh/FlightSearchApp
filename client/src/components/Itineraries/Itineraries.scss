@import '~bpk-mixins';

.Itineraries{
@media all and (max-width: 480px) {
    &__parent {
        height: 32px;
        width: 10%;
        border-radius: 3px;
        float: left;
        margin: 6% 34px 0 0;
    }
}
}