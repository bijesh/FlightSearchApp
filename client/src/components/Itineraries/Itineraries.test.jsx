import React from 'react';
import TestRenderer from 'react-test-renderer';

import Itineraries from './Itineraries';

const ItineraryDetailsData = [ {
    InBoundLeg : {
      ImageUrl:'EZY.png',
      DepartureTime: '2019-06-14T22:25:00',
      DepAirport:'LHR',
      ArrivalTime : '2019-06-14T22:25:00',
      ArvAirport : 'EDI',
      Duration:100,
      Stops : 1
    },
    OutBoundLeg : {
      ImageUrl:'EZY.png',
      DepartureTime: '2019-06-14T22:25:00',
      DepAirport:'LHR',
      ArrivalTime : '2019-06-14T22:25:00',
      ArvAirport : 'EDI',
      Duration:100,
      Stops : 1
    },
    BookingDetails : {
      price:100,
      agent: "EZY"
    }
  },
  {
    InBoundLeg : {
      ImageUrl:'BA.png',
      DepartureTime: '2019-06-14T22:25:00',
      DepAirport:'LHR',
      ArrivalTime : '2019-06-14T22:25:00',
      ArvAirport : 'EDI',
      Duration:100,
      Stops : 1
    },
    OutBoundLeg : {
      ImageUrl:'BA.png',
      DepartureTime: '2019-06-14T22:25:00',
      DepAirport:'LHR',
      ArrivalTime : '2019-06-14T22:25:00',
      ArvAirport : 'EDI',
      Duration:100,
      Stops : 1
    },
    BookingDetails : {
      price:100,
      agent: "BA"
    }
  }
];

describe('Itineraries', () => {
  it('should render correctly', () => {
    const tree = TestRenderer.create(<Itineraries itineraryDetails={ItineraryDetailsData} />).toJSON();

    expect(tree).toMatchSnapshot();
  });
});
