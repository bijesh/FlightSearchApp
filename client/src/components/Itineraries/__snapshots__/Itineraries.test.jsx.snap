// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Itineraries should render correctly 1`] = `
Array [
  <div
    className="bpk-card bpk-card--padded Itinerary__main"
    role="button"
>
    <div
        className="Itinerary__outbound"
    >
        <div>
            <div
                className="Leg__img-parent"
            >
                <img
                    alt=""
                    src="EZY.png"
                />
            </div>
            <div
                className="Leg__origin"
            >
                <p
                    className="bpk-text bpk-text--base"
                >
                    22
                </p>
                <p
                    className="bpk-text bpk-text--base Leg__origin__place"
                >
                    LHR
                </p>
            </div>
            <div
                className="Leg__arrow"
            >
                <svg
                    className="Leg__rightArrowIcon"
                    height="24"
                    style={
                        Object {
                            "height": "1.5rem",
                            "width": "1.5rem",
                          }
                    }
                    width="24"
                    xmlns="http://www.w3.org/2000/svg"
                >
                    <path
                        d="M14.4 19.5l5.7-5.3c.4-.4.7-.9.8-1.5.1-.3.1-.5.1-.7s0-.4-.1-.6c-.1-.6-.4-1.1-.8-1.5l-5.7-5.3c-.8-.8-2.1-.7-2.8.1-.8.8-.7 2.1.1 2.8l2.7 2.5H5c-1.1 0-2 .9-2 2s.9 2 2 2h9.4l-2.7 2.5c-.5.4-.7 1-.7 1.5s.2 1 .5 1.4c.8.8 2.1.8 2.9.1z"
                    />
                </svg>
            </div>
            <div
                className="Leg__destination"
            >
                <p
                    className="bpk-text bpk-text--base"
                >
                    22
                </p>
                <p
                    className="bpk-text bpk-text--base Leg__origin__place"
                >
                    EDI
                </p>
            </div>
            <div
                className="Leg__length"
            >
                <div
                    className="Leg__duration"
                >
                    100
                </div>
                <div
                    className="Leg__stops"
                >
                    1 Stop
                </div>
            </div>
        </div>
    </div>
    <div
        className="Itinerary__inbound"
    >
        <div>
            <div
                className="Leg__img-parent"
            >
                <img
                    alt=""
                    src="EZY.png"
                />
            </div>
            <div
                className="Leg__origin"
            >
                <p
                    className="bpk-text bpk-text--base"
                >
                    22
                </p>
                <p
                    className="bpk-text bpk-text--base Leg__origin__place"
                >
                    LHR
                </p>
            </div>
            <div
                className="Leg__arrow"
            >
                <svg
                    className="Leg__rightArrowIcon"
                    height="24"
                    style={
                        Object {
                            "height": "1.5rem",
                            "width": "1.5rem",
                          }
                    }
                    width="24"
                    xmlns="http://www.w3.org/2000/svg"
                >
                    <path
                        d="M14.4 19.5l5.7-5.3c.4-.4.7-.9.8-1.5.1-.3.1-.5.1-.7s0-.4-.1-.6c-.1-.6-.4-1.1-.8-1.5l-5.7-5.3c-.8-.8-2.1-.7-2.8.1-.8.8-.7 2.1.1 2.8l2.7 2.5H5c-1.1 0-2 .9-2 2s.9 2 2 2h9.4l-2.7 2.5c-.5.4-.7 1-.7 1.5s.2 1 .5 1.4c.8.8 2.1.8 2.9.1z"
                    />
                </svg>
            </div>
            <div
                className="Leg__destination"
            >
                <p
                    className="bpk-text bpk-text--base"
                >
                    22
                </p>
                <p
                    className="bpk-text bpk-text--base Leg__origin__place"
                >
                    EDI
                </p>
            </div>
            <div
                className="Leg__length"
            >
                <div
                    className="Leg__duration"
                >
                    100
                </div>
                <div
                    className="Leg__stops"
                >
                    1 Stop
                </div>
            </div>
        </div>
    </div>
    <div
        className="Price__footer"
    >
        <div
            className="Price__airline"
        >
            <div
                className="Price__airline__price"
            >
                Â£
                100
            </div>
            <div
                className="Price__airline__name"
            >
                EZY
            </div>
        </div>
        <div
            className="Price__button"
        >
            <button
                className="bpk-button bpk-button--large Price__airline__select"
                disabled={false}
                onClick={null}
                type="button"
            >
                Select
            </button>
        </div>
    </div>
</div>,
  <div
    className="bpk-card bpk-card--padded Itinerary__main"
    role="button"
>
    <div
        className="Itinerary__outbound"
    >
        <div>
            <div
                className="Leg__img-parent"
            >
                <img
                    alt=""
                    src="BA.png"
                />
            </div>
            <div
                className="Leg__origin"
            >
                <p
                    className="bpk-text bpk-text--base"
                >
                    22
                </p>
                <p
                    className="bpk-text bpk-text--base Leg__origin__place"
                >
                    LHR
                </p>
            </div>
            <div
                className="Leg__arrow"
            >
                <svg
                    className="Leg__rightArrowIcon"
                    height="24"
                    style={
                        Object {
                            "height": "1.5rem",
                            "width": "1.5rem",
                          }
                    }
                    width="24"
                    xmlns="http://www.w3.org/2000/svg"
                >
                    <path
                        d="M14.4 19.5l5.7-5.3c.4-.4.7-.9.8-1.5.1-.3.1-.5.1-.7s0-.4-.1-.6c-.1-.6-.4-1.1-.8-1.5l-5.7-5.3c-.8-.8-2.1-.7-2.8.1-.8.8-.7 2.1.1 2.8l2.7 2.5H5c-1.1 0-2 .9-2 2s.9 2 2 2h9.4l-2.7 2.5c-.5.4-.7 1-.7 1.5s.2 1 .5 1.4c.8.8 2.1.8 2.9.1z"
                    />
                </svg>
            </div>
            <div
                className="Leg__destination"
            >
                <p
                    className="bpk-text bpk-text--base"
                >
                    22
                </p>
                <p
                    className="bpk-text bpk-text--base Leg__origin__place"
                >
                    EDI
                </p>
            </div>
            <div
                className="Leg__length"
            >
                <div
                    className="Leg__duration"
                >
                    100
                </div>
                <div
                    className="Leg__stops"
                >
                    1 Stop
                </div>
            </div>
        </div>
    </div>
    <div
        className="Itinerary__inbound"
    >
        <div>
            <div
                className="Leg__img-parent"
            >
                <img
                    alt=""
                    src="BA.png"
                />
            </div>
            <div
                className="Leg__origin"
            >
                <p
                    className="bpk-text bpk-text--base"
                >
                    22
                </p>
                <p
                    className="bpk-text bpk-text--base Leg__origin__place"
                >
                    LHR
                </p>
            </div>
            <div
                className="Leg__arrow"
            >
                <svg
                    className="Leg__rightArrowIcon"
                    height="24"
                    style={
                        Object {
                            "height": "1.5rem",
                            "width": "1.5rem",
                          }
                    }
                    width="24"
                    xmlns="http://www.w3.org/2000/svg"
                >
                    <path
                        d="M14.4 19.5l5.7-5.3c.4-.4.7-.9.8-1.5.1-.3.1-.5.1-.7s0-.4-.1-.6c-.1-.6-.4-1.1-.8-1.5l-5.7-5.3c-.8-.8-2.1-.7-2.8.1-.8.8-.7 2.1.1 2.8l2.7 2.5H5c-1.1 0-2 .9-2 2s.9 2 2 2h9.4l-2.7 2.5c-.5.4-.7 1-.7 1.5s.2 1 .5 1.4c.8.8 2.1.8 2.9.1z"
                    />
                </svg>
            </div>
            <div
                className="Leg__destination"
            >
                <p
                    className="bpk-text bpk-text--base"
                >
                    22
                </p>
                <p
                    className="bpk-text bpk-text--base Leg__origin__place"
                >
                    EDI
                </p>
            </div>
            <div
                className="Leg__length"
            >
                <div
                    className="Leg__duration"
                >
                    100
                </div>
                <div
                    className="Leg__stops"
                >
                    1 Stop
                </div>
            </div>
        </div>
    </div>
    <div
        className="Price__footer"
    >
        <div
            className="Price__airline"
        >
            <div
                className="Price__airline__price"
            >
                Â£
                100
            </div>
            <div
                className="Price__airline__name"
            >
                BA
            </div>
        </div>
        <div
            className="Price__button"
        >
            <button
                className="bpk-button bpk-button--large Price__airline__select"
                disabled={false}
                onClick={null}
                type="button"
            >
                Select
            </button>
        </div>
    </div>
</div>,
]
`;
